feat(formatter): support ignore comment

Dunqing opened the pull request:
Support ignores comments like `// prettier-ignore` and `/* prettier-ignore */ to disable formatting, but do not add support for something like `oxc-formatter-ignore` yet.

Unfortunately, this less important feature slightly hurts performance, as we need to check whether there is an ignored comment at the start of each node. I will come back to improve performance when I finish the last typecast comment feature.

Boshen commented:
### Merge activity

* **Aug 29, 3:34 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
feat(formatter): support ignore comment (#13322)

Support ignores comments like `// prettier-ignore` and `/* prettier-ignore */ to disable formatting, but do not add support for something like `oxc-formatter-ignore` yet.

Unfortunately, this less important feature slightly hurts performance, as we need to check whether there is an ignored comment at the start of each node. I will come back to improve performance when I finish the last typecast comment feature.
